document GROUP BY (#1607)
neo4j/docs-cypher: modules/ROOT/pages/functions/aggregating.adoc

Reviewer: Would it be useful if this paper would also point to `GROUP BY` in some way? Like say in some fitting place something like "For more precise control of the grouping keys the use of the GROUP BY clause is recommended" and/or in

```
*Grouping keys* are non-aggregating expressions that are used to group the values going into the aggregating functions.
For example, given the following query containing two return expressions, `n` and `+count(*)+`:
```

add

```
*Grouping keys* either given implicit or explicitly.
Implicit grouping keys are non-aggregating expressions that are used to group the values going into the aggregating functions.
For example, given the following query containing two return expressions, `n` and `+count(*)+`:
Explicit grouping keys are given by the `GROUP BY` clause [link].
```

Something like that. A user who looks up aggregation should here about `GROUP BY`.

Both are input arguments to the division expression.

*Grouping keys* are non-aggregating expressions that are used to group the values going into the aggregating functions.

> Grouping keys are implicit in Cypher but can be controlled explicitly with the xref:clauses/group-by.adoc[`GROUP BY` clause].

Reviewer: The added test helps. But sounds like grouping keys can be both implicit and explicit at the same time. They are either–or. Without GROUP BY, they are implicit (implicitly identified by their lack of aggregate function). With GROUP BY, they are explicit (explicitly identified/stated in the GROUP BY).

Here is a suggestion:

```suggestion
Grouping keys are either explicitly listed in the xref:clauses/group-by.adoc[`GROUP BY` clause] or implicit if no `GROUP BY` clause is given.
```

Grouping keys themselves can be complex expressions.
Cypher only recognizes a sub-expression in aggregating expressions as a grouping key if the grouping key is:

- A variable - e.g. the `p` in `RETURN p, p.age - max(f.age)`.
- A property access - e.g. the `p.age` in `RETURN p.age, p.age - max(f.age)`.
- A map access - e.g. the `p.age` in `WITH {name:'Keanu Reeves', age:58} AS p RETURN p.age, p.age - max(p.age)`.

These are called _recognizable grouping key expressions_.

> If you need more complex grouping keys as operands in aggregating expressions, you can always project them with `AS` in `RETURN` and `WITH`.

Reviewer 2: ```suggestion
If you need more complex grouping keys as operands in aggregating expressions, you can project them with `AS` in `RETURN` and `WITH`.
```
